Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -17,296 +17,526 @@
 
-@interface CGNSContext : _CGLifetimeBridgingType
-@end
+    inline ComPtr<ID2D1Factory> Factory() {
+        ComPtr<ID2D1Factory> factory;
+        _impl.renderTarget->GetFactory(&factory);
+        return factory;
+    }
+};
 
-@implementation CGNSContext
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wobjc-missing-super-calls"
-- (void)dealloc {
-    delete (__CGContext*)self;
+#pragma region CFRuntimeClass
+static Boolean __CGContextEqual(CFTypeRef cf1, CFTypeRef cf2) {
+    return FALSE;
 }
-#pragma clang diagnostic pop
-@end
 
-__CGContext::__CGContext(CGImageRef pDest) {
-    contextCount++;
+static CFHashCode __CGContextHash(CFTypeRef cf) {
+    return (CFHashCode)cf;
+}
 
-#ifdef DEBUG_CONTEXT_COUNT
-    TraceVerbose(TAG, L"contextCount: %d", contextCount);
-#endif
-    object_setClass((id) this, [CGNSContext class]);
-    scale = 1.0f;
-    _backing = pDest->Backing()->CreateDrawingContext(this);
+static CFStringRef __CGContextCopyDescription(CFTypeRef cf) {
+    return CFStringCreateWithFormat(kCFAllocatorSystemDefault, NULL, CFSTR("<CGContext %p>"), cf);
 }
 
-__CGContext::~__CGContext() {
-    contextCount--;
+static void __CGContextInit(CFTypeRef cf) {
+    CGContextRef context = (CGContextRef)cf;
+    struct __CGContext* mutableContext = const_cast<struct __CGContext*>(context);
+    new (&mutableContext->_impl) __CGContext::__CGContextImpl();
+}
 
-    delete _backing;
+static void __CGContextDeallocate(CFTypeRef cf) {
+    CGContextRef context = (CGContextRef)cf;
+    context->_impl.~__CGContextImpl();
 }
 
-CGContextImpl* __CGContext::Backing() {
-    return _backing;
+template <typename T, typename PT = typename std::remove_pointer<T>::type*, typename B = decltype(std::declval<PT>()->_base)>
+static PT _CFRuntimeCreateInstance(CFTypeID typeID, CFAllocatorRef allocator = kCFAllocatorDefault) {
+    return static_cast<PT>(
+        const_cast<void*>(_CFRuntimeCreateInstance(allocator, typeID, sizeof(typename std::remove_pointer<T>::type) - sizeof(B), nullptr)));
 }
 
 /**
  @Status Interoperable
 */
-void CGContextSetBlendMode(CGContextRef ctx, CGBlendMode mode) {
-    ctx->Backing()->CGContextSetBlendMode(mode);
+CFTypeID CGContextGetTypeID() {
+    static CFTypeID __kCGContextTypeID = _kCFRuntimeNotATypeID;
+    static const CFRuntimeClass __CGContextClass = { 0,    "CGContext", __CGContextInit,           NULL, __CGContextDeallocate, NULL,
+                                                     NULL, NULL,        __CGContextCopyDescription };
+
+    static dispatch_once_t initOnce = 0;
+    dispatch_once(&initOnce, ^{
+        __kCGContextTypeID = _CFRuntimeRegisterClass(&__CGContextClass);
+    });
+    return __kCGContextTypeID;
+}
+#pragma endregion
+
+static const wchar_t* TAG = L"CGContext";
+
+static void __CGContextInitializeWithRenderTarget(CGContextRef context, ID2D1RenderTarget* renderTarget) {
+    context->_impl.renderTarget = renderTarget;
+    CGContextSetRGBFillColor(context, 0, 0, 0, 0);
```

Comment:
>CGContextSetRGBFillColor [](start = 4, length = 24)

Should the default be white on black, or does this black on black default behavior match reference platform?